### ISSUE TITLE
Tensor from blob complaining about wrong CUDA device

### ISSUE THREAD

**Issue by xsacha** [pytorch/pytorch#19007]
## 🐛 Bug
I get the following error in the latest nightly when attempting to convert a GPU blob from OpenCV to torch:

> Attempted to set the storage of a tensor on device cuda:0 to a storage on different device cuda.  This is no longer allowed; the devices must match.

Code that creates this issue:
```
     std::vector<int64_t> dims = {image.rows, image.cols, 3};
     long long step = image.step / sizeof(float);
     std::vector<int64_t> strides = {step, 3, 1};
     auto tensor = torch::from_blob(image.data, dims, strides, deleter, torch::kCUDA);
```

**Note: I only have a single CUDA GPU on this system!**
The default cuda device is negative 1, which does not match the actual cuda device of 0. However, this code previously worked fine.

## To Reproduce

1. Download latest Windows nightly
2. Try converting a tensor from blob as torch::kCUDA

## Expected behavior

Successfully create a tensor from existing blob data

## Environment

 - PyTorch Version (e.g., 1.0): 1.1.0-pre
 - OS (e.g., Linux): Windows 64-bit
 - How you installed PyTorch (`conda`, `pip`, source): nightly
 - Build command you used (if compiling from source): N/A
 - Python version: N/A
 - CUDA/cuDNN version: 10.0/7.5
 - GPU models and configuration: 1060
 - Any other relevant information: Only in most recent nightly

## Additional Context
Caused from this commit: https://github.com/pytorch/pytorch/commit/8732a1b42ea46fc251369bab4f29332d5e35959e#diff-aa7cacda9b64a355ec49c7298cca1726

Changing the code to:
```
auto tensor = torch::from_blob(image.data, dims, strides, deleter, torch::device(torch::Device(torch::kCUDA, 0)));
```

resolves the issue.

**Comment by xsacha** [comment]
Not 100% sure this is a bug because the code technically won't work in a multi-GPU configuration.
It has forced me to correctly pass through the GPU device.

**Comment by umanwizard** [comment]
I doubt this is Windows specific so removing that label. Can someone comment on whether this is a bug or intended behavior

**Comment by ezyang** [comment]
cc @gchanan

**Comment by ezyang** [comment]
We should at the very least give a better error message here.

**Comment by gchanan** [comment]
Looks like this was actually broken by https://github.com/pytorch/pytorch/pull/18779.

**Comment by gchanan** [comment]
Reassigning to @li-roy.

**Comment by li-roy** [comment]
closed by #19094
